src/Web/Pages.cpp

```diff
@@ -257,6 +260,24 @@ void Pages::init(AsyncWebServer& srv)
         ++idx;
     }
 
+    (void)srv.on("/change-partition", HTTP_GET, [](AsyncWebServerRequest* request) {
+        if (false == setFactoryPartitionActive())
+        {
+            request->send(500, "text/plain", "Failed to switch partition.");

```

Comment:
It would be good to send more error information here instead of  a generic failed.  The user may not have access to the logs produced by  setFactoryPartitionActive().